bp-admin-only-profile-fields.php

```diff
@@ -144,30 +146,124 @@ public function custom_visibility_levels( $levels ) {
 			'id'    => 'hidden',
 			'label' => __( 'Hidden', 'bp_admin_only_profile_fields' )
 		);
+		$levels['internal'] = array(
+			'id'    => 'internal',
+			'label' => __( 'Admin Internal', 'bp_admin_only_profile_fields' )
+		);
+		$levels['adminedit'] = array(
+			'id'    => 'adminedit',
+			'label' => __( 'Admin Editable', 'bp_admin_only_profile_fields' )
+		);
+
 
 		return $levels;
 	}
 
 	/**
-	 * Hide our hidden fields.
+	 * Append 'hidden' to the visibility levels for this user pair.
 	 *
-	 * @since  1.0
+	 * @since  1.2
 	 *
-	 * @param array $hidden_fields
+	 * @param array $hidden_levels
 	 * @param int   $displayed_user_id
 	 * @param int   $current_user_id
 	 *
 	 * @return array
 	 */
-	public function hide_hidden_fields( $hidden_fields, $displayed_user_id, $current_user_id ) {
-
-		$hidden_fields = bp_xprofile_get_fields_by_visibility_levels( $displayed_user_id, array( 'hidden' ) );
+	public function append_hidden_level( $hidden_levels, $displayed_user_id, $current_user_id ) {
 
 		if ( ! current_user_can( apply_filters( 'bp_admin_only_profile_fields_cap', 'manage_options' ) ) ) {
-			return $hidden_fields;
+
+			// Current user is non-admin
+			$hidden_levels[] = 'hidden';
+
+			if ( empty( $current_user_id ) ) {
+
+				// Current user is not logged in
+				$hidden_levels[] = 'internal';
+
+			} else {
+
+				if ( $displayed_user_id != $current_user_id ) {
+
+					// Not viewing own profile
+					$hidden_levels[] = 'internal';
+
+				} else {
+
+					if ( bp_is_profile_edit() ) {
```

Comment:
Thanks @A5hleyRich, I've replaced the deprecated function and pushed my changes. Cheers